### DISCUSSION TITLE
Finding a good layout - X-ray Room

### DISCUSSION THREAD

**Discussion by ageekhere** [CorsixTH/CorsixTH#2418] [General Diagnostics]
I am exploring good design layouts for each room, in this post I am focusing on the X-ray Room

**The Goal**
Keeping the room as small as possible while keeping the doctor as happy as possible(avoid being unhappy and wanting a pay rise) while optimizing turnaround time for handyman plant watering and patient walk times.

**Version 1.00**
![Screenshot 2023-09-28 084032](https://github.com/CorsixTH/CorsixTH/assets/13173133/fb046a8f-5352-429a-b11f-074cb4b6fad2)

Items
1x X-ray
1x Radiation Shield
1x Bin
1x Fire extinguisher
1x Plant
2x Radiator

**Summary**
The X-ray Room has 2 main items, the X-ray and the Radiation Shield with the X-ray's location being the most important. The patient 
 enters the X-ray from the right so we want the X-ray entry to be as close as we can to the door. Version 1.00 positions the X-ray as close to the door as possible, even though the patient does need to make a turn i think the walk path is optimized. Plant is next to door and bin is in patient walk path.  Fire extinguisher is center of the room and 2 Radiators are on either side of the room.

Lets start the decision, can this design be improved and if so how?

**Version 1.01**
![Screenshot 2023-09-30 121833](https://github.com/CorsixTH/CorsixTH/assets/13173133/198b1298-b709-4e21-828a-d54162dc1a99)
Possible optimization of the Radiation Shield.

**Comment by MandrakeHorse** [comment]
Here's my proposed layout: not only does it optimise the patient's journey to the X-Ray, but it also aims to optimise the doctor's journey to the Radiation Shield - he can simply walk in and sit down at the console. Fortunately, the radiation shields are purely cosmetic and the doctor will suffer no consequences from repeated x-ray exposure! (Ignore the graphical glitch in the corner - this is a known issue: https://github.com/CorsixTH/CorsixTH/issues/403)
![image](https://github.com/CorsixTH/CorsixTH/assets/142453312/49e04b13-d94d-4a40-9fc1-b0a9bd0980cc)

**Reply by ageekhere** [reply]
Design 1.00 has 4 squares from door to X-ray where this design has 5 squares so 1 more square the patient has to walk. Though i have an idea for 1.01.

**Comment by ARGAMX** [comment]
Thanks, I like the idea of Finding a good layout for each room.

I want to add that the count of extinguishers does matter.
In CorsixTH the fourth and each next extinguishers does not matter. Only ~~first 3~~ first 4.
So it makes sense to have at least 3 fire extinguishers.
They do not block the way and there easy to place them in the X-ray room because of big space of that room.
Why this layout suggests only 1 extinguisher?
The same applies to the other rooms with machine.

**Reply by ageekhere** [reply]
This is a good finding, at the time it was not clear to me the logic on the number of fire extinguishers to use
